lib/Importer/Caffe2.cpp

```diff
@@ -165,8 +178,8 @@ void caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
   if (typeName == "MaxPool" || typeName == "AveragePool") {
     // Load the inputs:
     auto *in = getOrCreateNodeByName(op.input(0));
-    int stride = loadInt(dict["stride"]);
-    size_t kernel = loadInt(dict["kernel"]);
+    int stride = getSizeHW(dict, "stride", 1);
+    unsigned kernel = getSizeHW(dict, "kernel", 0);
```

Comment:
Why unsigned/int and not size_t for all of the kernel/stride/pad/group? The ConvolutionNode etc. all use size_t.

Reply:
I don't have a good understanding of how we differentiate between `unsigned` and `size_t`. Isn't it better to always use a type with stable (spec'd) size?

Reply:
@nadavrot Yes, I removed all size_t usages from OpenCL kernel APIs to make sure we use a stable size for our unsigned types. We should probably migrate to using uint64_t, but may with a nicer name alias.

Reply:
@opti-mix Let's do it! (after checking with everybody in the team)

Reply:
I'd go for int64_t. We may see -1 occasionally

Reply:
@artemrakhov Where do we use -1?

Reply:
For reshape, sometimes it's possible to have {5, -1} written in a proto as new shape. If input's shape was {100}, then -1 will be automatically deduced to 20, so that the size is preserved: {5, 20}.

Very few other operators (e.g. TopK) support axis=-1. In this case it means "take the last axis".

So far all these cases could be resolved on Importer stage, before going to our Graph.

Reply:
@artemrakhov Okay. In that case I think that we should leave the negative values inside the loader. I think that the dimensions of tensors should be strictly positive in the compiler itself.